electrum/submarine_swaps.py

```diff
@@ -1515,7 +1528,15 @@ async def check_direct_messages(self):
             if 'reply_to' in content:
                 self.dm_replies[content['reply_to']].set_result(content)
             elif self.sm.is_server and 'method' in content:
-                await self.handle_request(content)
+                try:
+                    await self.handle_request(content)
+                except Exception as e:
+                    self.logger.exception(f"failed to handle request: {content}")
+                    error_response = json.dumps({
+                        "error": str(e)[:100],
+                        "reply_to": event.id,
+                    })
+                    await self.send_direct_message(event.pubkey,[], error_response)
```

Comment:
what is this for? it two servers are talking to each other, could it trigger an infinite loop?